core/src/main/java/com/spotify/metrics/core/SemanticMetricRegistry.java

```diff
@@ -69,8 +71,13 @@ public SemanticMetricRegistry(final ConcurrentMap<MetricId, Metric> metrics) {
     public SemanticMetricRegistry() {
         // This is only for backward compatibility purpose. After removing the "buildMap" method
         // we should call this(new ConcurrentHashMap<MetricId, Metric>()) instead.
-        this.metrics = buildMap();
+        this(new ConcurrentHashMap<MetricId, Metric>(), () -> new ExponentiallyDecayingReservoir());
+    }
+
+    public SemanticMetricRegistry(final ConcurrentMap<MetricId, Metric> metrics, final Supplier<Reservoir> defaultReservoirSupplier) {
```

Comment:
Nice, so this will allow us to pass in our own `Reservoir` when constructing a `SemanticMetricRegistry` right?

Reply:
Yep!